src/main.py

```diff
@@ -61,7 +61,7 @@
 # init vector database
 vectordb = hosted_database.build_database(
     embedding, 
-    docs, 
+    [doc for doc in docs],

```

Comment:
The markdown loader now does lazy loading.
This means that it only returns an iterator instead of an actual list of documents.

Unfortunately langchain doesn't support the lazy loading of documents (as far as I could see)

Note:
Lazy loading improves the memory handling because we only have to compute/load the document when requested and would only need to live in memory until consumed